components/brave_rewards/resources/extension/brave_rewards/background/reducers/rewards_panel_reducer.ts

```diff
@@ -102,9 +88,15 @@ export const rewardsPanelReducer = (state: RewardsExtension.State | undefined, a
         }
 
         publishers[id] = {
-          tabUrl: tab.url
+          tabUrl: tab.url,
+          tabId: tab.id
         }
+      } else if (publisher && publisher.tabUrl === tab.url && publisher.tabId !== tab.id && validKey) {
```

Comment:
nit: can we break this one up a little bit :)

Reply:
done, let me know if this is what you had in mind